code/modules/client/preferences_savefile.dm

```diff
@@ -49,6 +49,9 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		pda_style = "mono"
 	if(current_version < 20)
 		pda_color = "#808000"
+	if((current_version < 21) && S["feature_ethcolor"] && (S["feature_ethcolor"] == "#9c3030"))
+		WRITE_FILE(S["feature_ethcolor"]	, "9c3030")
```

Comment:
This proc is called after the character file is loaded into the pref datum, so you should be reading and changing the datum vars not the savefile values.

Also the system should save the save file when it does the version conversion, it's not currently doing that. that can be fix in another pr.